config/default/manager_webhook_patch.yaml

```diff
@@ -17,23 +11,20 @@ spec:
       - name: manager
         env:
         - name: WEBHOOK_SERVICE_NAMESPACE
-          value: $(WEBHOOK_SERVICE_NAMESPACE)
+          value: WEBHOOK_SERVICE_NAMESPACE_PLACEHOLDER
         - name: WEBHOOK_SERVICE_NAME
-          value: $(WEBHOOK_SERVICE_NAME)
+          value: WEBHOOK_SERVICE_NAME_PLACEHOLDER
         - name: WEBHOOK_SECRET_NAMESPACE
-          value: $(WEBHOOK_SECRET_NAMESPACE)
+          value: WEBHOOK_SECRET_NAMESPACE_PLACEHOLDER
         - name: WEBHOOK_SECRET_NAME
-          value: $(WEBHOOK_SECRET_NAME)
+          value: WEBHOOK_SECRET_NAME_PLACEHOLDER
         - name: WEBHOOK_SECRET_VOLUME_MOUNT_PATH
           value: WEBHOOK_SECRET_VOLUME_MOUNT_PATH_VALUE
         - name: WEBHOOK_SERVICE_CONTAINER_PORT
           value: WEBHOOK_SERVICE_CONTAINER_PORT_STRING
         ports:
-        # This value is used as the int constant for webhook port across our YAML
-        # String constant for the same port is webhooks.vmoperator.vmware.com/service-container-port above. The two should be consistent.
-        # Note also there's an assumption in kustomization.yaml that this is ports[0]
         - containerPort: 9878
-          name: webhook-server
+          name: https
```

Comment:
This name is already used for the kube-rbac-proxy container port: https://github.com/vmware-tanzu/vm-operator/blob/main/config/default/manager_auth_proxy_patch.yaml#L25

I believe this would cause unpredictable traffic routing, since both metrics and webhook Services have the same label selector and target port name?
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    prometheus.io/port: "9848"
  labels:
    control-plane: controller-manager
  name: vmware-system-vmop-controller-manager-metrics-service
  namespace: vmware-system-vmop
spec:
  ports:
  - name: https
    port: 9848
    targetPort: https
  selector:
    control-plane: controller-manager
---
apiVersion: v1
kind: Service
metadata:
  name: vmware-system-vmop-webhook-service
  namespace: vmware-system-vmop
spec:
  ports:
  - name: https
    port: 443
    targetPort: https
  selector:
    control-plane: controller-manager
```

Reply:
Good call. I'll update.